### DISCUSSION TITLE
Additional methods to put code blocks and their outputs side by side, other than what's shown below?

### DISCUSSION THREAD

**Discussion by hopezh** [quarto-dev/quarto-cli#8966] [Q&A]
### Description

I summarized three methods I learned to put code blocks, and their outputs, side by side in multiple columns in [this notes made from quarto](https://hopezh.github.io/oatNotes/side_by_side_codeblock.html), with source of references cited, too. 

1. using `{.column-screen-insert}`
2. using `{layout-ncol='3'}`
3. using `{.columns}` and `{.column width="30%"}`

I'd like to learn additional methods to achieve this effect in quarto.

Thanks. 

![2024-03-04 11 50 37](https://github.com/quarto-dev/quarto-cli/assets/23435810/6a2771cf-bc4b-496c-a7a1-5b69ceb14677)

**Comment by mcanouil** [comment]
`{.column-screen-insert}` is not a method to put things side by side. It's a layout option to change body width.

columns/layout fenced divs are not exactly the same as "columns" (Pandoc) does not collapse code cells together while "layout" (Quarto) does.

Note that for `revealjs`, you can directly output next to the code (or to the next slide), see <https://quarto.org/docs/reference/formats/presentations/revealjs.html#slide-content>.

I am not completely sure what you expect here with your question besides what is already in the documentation about putting things side by side (no matter that those things are code, figures, text, etc.).
"Additional methods" would be raw blocks of the underlying format language: HTML, OpenXML, LaTeX, Typst, etc. For those, you'll have to learn the languages, read their documentation, etc.
